### PR TITLE
[KeyVault] Remove references to six package

### LINKED ISSUE #22373: [Key Vault] Remove references to six package
Now that support for Python 2.7 has been dropped (https://github.com/Azure/azure-sdk-for-python/pull/22367), we can stop using the `six` package.

### PR DESCRIPTION
**Pull request by ponopono0322**
# Description

Fix #22373.
Removed `six` package

# All SDK Contribution checklist:
- [x] **The pull request does not introduce [breaking changes]**
- [x] **CHANGELOG is updated for new features, bug fixes or other significant changes.**
- [x] **I have read the [contribution guidelines](https://github.com/Azure/azure-sdk-for-python/blob/main/CONTRIBUTING.md).**

## General Guidelines and Best Practices
- [x] Title of the pull request is clear and informative.
- [x] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).

### [Testing Guidelines](https://github.com/Azure/azure-sdk-for-python/blob/main/CONTRIBUTING.md##building-and-testing)
- [x] Pull request includes test coverage for the included changes.

### CODE REVIEW

**Comment by ponopono0322** [comment]
Summary of changes

1. `.moves.urllib_parse`
Following [docs](https://six.readthedocs.io/#module-six.moves), same as using `urllib`.
```python
link = "https://azure.microsoft.com/en-us/"
# Before
import six
q = six.moves.urllib_parse.urlparse(link)
print(p)
# >>> ParseResult(scheme='https', netloc='azure.microsoft.com', path='/en-us/', params='', query='', fragment='')

# After
from urllib import parse
p = parse.urlparse(link)
print(q)
# >>> ParseResult(scheme='https', netloc='azure.microsoft.com', path='/en-us/', params='', query='', fragment='')
```

2. `six.raise_from`
Following [docs](https://six.readthedocs.io/#six.raise_from), `six.raise_from(exc_value, exc_value_from)` same as `raise exc_value from exc_value_from`
```python
# Before
from six import raise_from
try:
    zero_div = 1 / 0
except Exception as ex:
    raise_from(ValueError("zero div"), ex, )
# >>> ValueError: zero div

# After
try:
    zero_div = 1 / 0
except Exception as ex:
    raise ValueError("zero div") from ex
# >>> ValueError: zero div
```

3. `six.with_metaclass`
```python
from enum import Enum, EnumMeta
class _CaseInsensitiveEnumMeta(EnumMeta):
    def __getitem__(self, name):
        return super().__getitem__(name.upper())

    def __getattr__(cls, name):
        try:
            return cls._member_map_[name.upper()]
        except KeyError:
            raise AttributeError(name)

# Before
from six import with_metaclass
class DeletionRecoveryLevel(with_metaclass(_CaseInsensitiveEnumMeta, str, Enum)):
    PURGEABLE = "Purgeable"
    RECOVERABLE_PURGEABLE = "Recoverable+Purgeable"
    RECOVERABLE = "Recoverable"
    RECOVERABLE_PROTECTED_SUBSCRIPTION = "Recoverable+ProtectedSubscription"
    CUSTOMIZED_RECOVERABLE_PURGEABLE = "CustomizedRecoverable+Purgeable"
    CUSTOMIZED_RECOVERABLE = "CustomizedRecoverable"
    CUSTOMIZED_RECOVERABLE_PROTECTED_SUBSCRIPTION = "CustomizedRecoverable+ProtectedSubscription"


print(type(DeletionRecoveryLevel), DeletionRecoveryLevel)
print(type(DeletionRecoveryLevel.PURGEABLE), DeletionRecoveryLevel.PURGEABLE)
# >>> <class '__main__._CaseInsensitiveEnumMeta'> <enum 'DeletionRecoveryLevel'>
# >>> <enum 'DeletionRecoveryLevel'> DeletionRecoveryLevel.PURGEABLE

# After
class DeletionRecoveryLevel(str, Enum, metaclass=_CaseInsensitiveEnumMeta):
    PURGEABLE = "Purgeable"
    RECOVERABLE_PURGEABLE = "Recoverable+Purgeable"
    RECOVERABLE = "Recoverable"
    RECOVERABLE_PROTECTED_SUBSCRIPTION = "Recoverable+ProtectedSubscription"
    CUSTOMIZED_RECOVERABLE_PURGEABLE = "CustomizedRecoverable+Purgeable"
    CUSTOMIZED_RECOVERABLE = "CustomizedRecoverable"
    CUSTOMIZED_RECOVERABLE_PROTECTED_SUBSCRIPTION = "CustomizedRecoverable+ProtectedSubscription"


print(type(DeletionRecoveryLevel), DeletionRecoveryLevel)
print(type(DeletionRecoveryLevel.PURGEABLE), DeletionRecoveryLevel.PURGEABLE)
# >>> <class '__main__._CaseInsensitiveEnumMeta'> <enum 'DeletionRecoveryLevel'>
# >>> <enum 'DeletionRecoveryLevel'> DeletionRecoveryLevel.PURGEABLE
```

4. `six.ensure_str`
Following [docs](https://six.readthedocs.io/#six.ensure_str), `six.ensure_str(s, encoding='utf-8', errors='strict'` same as `str(s)`
```python
r1 = """{"body": "something", "header": "anything"}"""
# Before
import six
print(type(six.ensure_str(r1)))
# >>> <class 'str'>

# After
print(type(str(r1)))
# >>> <class 'str'>
```

5. `six.string_types`
Following [docs](https://six.readthedocs.io/#six.string_types), `string_types` same as `str`
```python
# Before
import six
print("True") if isinstance("key", six.string_types) else print("False")
# >>> True

# After
print("True") if isinstance("key", str) else print("False")
# >>> True
```

6. `six. byte2int`
Following [docs](https://six.readthedocs.io/#six.byte2int), `six.byte2int(bs)` same as `bs[0]`.
```python
bs = bytes(5)

# Before
import six
print(bs, type(six.byte2int(bs)), six.byte2int(bs))
# >>> b'\x00\x00\x00\x00\x00' <class 'int'> 0

# After
print(bs, type(bs[0]), bs[0])
# >>> b'\x00\x00\x00\x00\x00' <class 'int'> 0
```

**Review by mccoyp — COMMENTED** [review]
Thank you so much for opening a PR! I'll follow up on the generated code's use of `six.with_metaclass` and will let you know if I can fix that in a separate PR.

File: sdk/keyvault/azure-keyvault-administration/azure/keyvault/administration/_generated/v7_2/models/_key_vault_client_enums.py [hunk 1]
```diff
@@ -7,11 +7,10 @@
 # --------------------------------------------------------------------------
 
 from enum import Enum
-from six import with_metaclass
 from azure.core import CaseInsensitiveEnumMeta
 
 
-class DataAction(with_metaclass(CaseInsensitiveEnumMeta, str, Enum)):
+class DataAction(str, Enum, metaclass=CaseInsensitiveEnumMeta):
```
[the remark this answers is not in the document]
Comment (ponopono0322) [reply sdk/keyvault/azure-keyvault-administration/azure/keyvault/administration/_generated/v7_2/models/_key_vault_client_enums.py]: I checked and uploaded the reverted commit.

File: sdk/keyvault/azure-keyvault-administration/azure/keyvault/administration/_enums.py [hunk 2]
```diff
@@ -3,21 +3,20 @@
 # Licensed under the MIT License.
 # ------------------------------------
 from enum import Enum
-from six import with_metaclass
 
 from azure.core import CaseInsensitiveEnumMeta
 
 # pylint:skip-file (avoids crash due to six.with_metaclass https://github.com/PyCQA/astroid/issues/713)
```
[the remark this answers is not in the document]
Comment (ponopono0322) [reply sdk/keyvault/azure-keyvault-administration/azure/keyvault/administration/_enums.py]: I uploaded a commit including files that were not mentioned here that removed the pylint:skip-file six.with_metaclass

**Comment by mccoyp** [comment]
Closing in favor of https://github.com/Azure/azure-sdk-for-python/pull/26150

**Pull request closed without merging**